step-templates/octopus-set-Octopus-releaese-notes-from-TFS-query.json

```diff
@@ -0,0 +1,77 @@
NullOrWhiteSpace($currentRelease.ReleaseNotes)){\n\twrite-host \"Release notes already filled in. $($currentRelease.ReleaseNotes)\"\n    Set-OctopusVariable -name \"ReleaseNotes\" -value $releaseNotes\n\texit;\n}\n\n\n#Get projectid\n$url = \"http://$($instance)/tfs/$($collection)/$($projectId)/_apis/projects/$($project)?&includeCapabilities=false&includeHistory=false&api-version=2.2\"\nwrite-host \"Invoking url= $($url)\"\n$projectresponse = Invoke-RestMethod $url -Method GET -Headers $headers\n\n$projectid = $projectresponse.id\nwrite-host \"projectid $($projectid)\"\n\n#Get the ID of the query to execute\n$queryResult = Invoke-RestMethod \"http://$($instance)/tfs/$($collection)/$($projectId)/_apis/$($pathquery)?$depth=1&api-version=2.2\" -Method GET -Headers $headers\nwrite-host \"queryResult $($queryResult)\"\n\n#https://{instance}/DefaultCollection/[{project}/]_apis/wit/wiql/{id}?api-version={version}\n$queryResult = Invoke-RestMethod \"http://$($instance)/tfs/$($collection)/$($projectId)/_apis/wit/wiql/$($queryResult.Id)?api-version=2.2\" -Method GET -Headers $headers\n\nWrite-Host \"Found $($queryResult.workItems.Count) number of workitems for query: ReleaseNotes$($releaseTag)\"\n\n$releaseNotes = \"**Work Items:**\"\n\n\nif($queryResult.workItems.Count -eq 0)\n{\n\tWrite-Host \"No work items for release\"\n\t$releaseNotes = \"`n no new work items\"\n}\nelse\n{\n\t#Create a list of ids\n\t$ids = [string]::Join(\"%2C\", ($queryResult.workItems.id))\n\n\t#Get all the work items\n\t$workItems = Invoke-RestMethod  \"http://$($instance)/tfs/$($collection)/_apis/wit/workItems?ids=$($ids)&fields=System.Title\" -Method GET -Headers $headers\n\n\tforeach($workItem in $workItems.value)\n\t{\n\t\t#Add line for each work item\n\t\t$releaseNotes = $releaseNotes + \"`n* [$($workItem.id)] (http://$($instance)/tfs/$($collection)/9981e67f-b27c-4628-b5cf-fba1d327aa07/_workitems/edit/$($workItem.id)) : $($workItem.fields.'System.Title')\"\n\t}\n\n}\n\n\n\n# Update the release notes for the current release\n$currentRelease.ReleaseNotes = $releaseNotes \nwrite-host \"Release notes $($currentRelease.ReleaseNotes)\"\nWrite-Host \"Updating release notes for $thisReleaseNumber`:`n`n\"\ntry {\n    $releaseUri = \"$baseUri/api/releases/$($currentRelease.Id)\"\n    write-host \"Release uri $($releaseUri)\"\n    $currentReleaseBody = $currentRelease | ConvertTo-Json\n    write-host \"Current release body $($currentReleaseBody)\"\n    $result = Invoke-RestMethod $releaseUri -Method Post -Headers $putReqHeaders -Body $currentReleaseBody -UseBasicParsing\n\twrite-host \"result $($result)\"\n} catch {\n    $result = $_.Exception.Response.GetResponseStream()\n    $reader = New-Object System.Io.StreamReader($result);\n    $responseBody = $reader.ReadToEnd();\n    Write-Host \"error $($responseBody)\"\n    throw \"Error occurred: $responseBody\"\n}\n\nSet-OctopusVariable -name \"ReleaseNotes\" -value $releaseNotes\nSet-OctopusVariable -name \"test\" -value \"$releaseNotes\""
```

Comment:
`Set-OctopusVariable -name "test" -value "$releaseNotes"`
Not sure if this was meant to be included?

Reply:
`write-host "octopusAPIKey: $($octopusAPIKey)"`
Now, if the API key is a sensitive variable that is being referenced this would be fine (as it would be masked), but if it's not we'd be writing sensitive information to the logs which in general is a bad idea. 

I'd say we remove this line.

Reply:
Same as above regarding `write-host "PAT: $($PAT)"`